### PR TITLE
♻️ Migrate cli to ESM

### PR DESCRIPTION
**Pull request by carloscuesta**
## Description

This PR updates the `gitmoji-cli` to an ESM so we can update a lot of the dependencies that we're using.

Thanks @aaditmshah for the help! 😊 

## Tests

- [x] All tests passed.

## Pending

- [x] Migrate `getConfiguration` to avoid using `require`

### CODE REVIEW

File: src/utils/configurationVault/getConfiguration.js [hunk 1]
```diff
@@ -30,11 +30,11 @@ const getConfiguration = (): { get: Function, set: Function } => {
     const packageJson = `${cwd()}/package.json`
     const configurationFile = `${cwd()}/.gitmojirc.json`
 
-    if (pathExists.sync(packageJson) && require(packageJson)?.gitmoji) {
+    if (pathExistsSync(packageJson) && require(packageJson)?.gitmoji) {
       return require(packageJson).gitmoji
```
Comment (carloscuesta) [inline src/utils/configurationVault/getConfiguration.js:34]: We need to migrate this to `readFileSync` as `require` can't be used

File: src/utils/configurationVault/getConfiguration.js [hunk 2]
```diff
@@ -30,11 +30,11 @@ const getConfiguration = (): { get: Function, set: Function } => {
...
     }
 
-    if (pathExists.sync(configurationFile) && require(configurationFile)) {
+    if (pathExistsSync(configurationFile) && require(configurationFile)) {
```
Comment (carloscuesta) [inline src/utils/configurationVault/getConfiguration.js:37]: We need to migrate this to `readFileSync` as `require` can't be used

File: src/utils/configurationVault/getConfiguration.js (hunk 1 above)
Comment (aaditmshah) [reply src/utils/configurationVault/getConfiguration.js:34]: Completed this task in #928.

File: src/utils/configurationVault/getConfiguration.js (hunk 2 above)
Comment (aaditmshah) [reply src/utils/configurationVault/getConfiguration.js:37]: Completed this task in #928.

**Pull request merged**